tests/common.py

```diff
@@ -77,6 +79,34 @@ def stop_hass():
     hass.start = start_hass
     hass.stop = stop_hass
 
+    orig_create_server = hass.loop.create_server
+
+    @wraps(orig_create_server)
+    def create_server(*args, **kwargs):
+        """Wrap the native create_server method to return test bindings.
+
+        If a test port is requested and there is a binding available, return
+        a new server with that binding instead.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        if len(args) >= 3:
+            port = args[2]
+        else:
+            port = kwargs.get('port')
+
+        if port in _TEST_INSTANCE_PORT_BINDS:
+            # Port binding has been precreated. Drop host/port from either
+            # *args or **kwargs, and add the created port binding.
+            args = args[:1]
```

Comment:
For reference, FYI: https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.AbstractEventLoop.create_server